### ISSUE TITLE
help ERROR: database already exists (pgsql://@/nominatim)

### ISSUE THREAD

**Issue by maxicabanillas78** [twain47/Nominatim#224]
good day to all
I need help please
I have installed map of brazil
but when I am installing Argentina estae I get Error

root@Maxi:~# ./Nominatim/utils/setup.php --osm-file argentina-latest.osm.pbf --all 2>&1 | tee import.txt
Create DB
ERROR: database already exists (pgsql://@/nominatim)
database already exists (pgsql://@/nominatim)

I need to install if or if urgent

**Comment by lonvia** [comment]
You cannot use setup.php to add more data to an existing database. For more information on how to do that, see https://help.openstreetmap.org/questions/15505/import-more-osm-files-in-to-nominatim

**Comment by mtmail** [comment]
You have three possibilities:

1. delete the existing database in postgres (`psql`, then `drop database nominatim`). Now you can only search Argentina because the Brazil data is no longer in the database

2. change the database name in `settings/local.php` for example to `nominatim-argentina`. The PHP frontend uses the same configuration file. Now you can search Argentina. You can still search Brazil when you change the configuration file again. You can't search both at the same time.

3. merge the input files `brazil-latest.osm.pbf` and `argentina-latest.osm.pbf`. Delete the `nominatim` database and input the new file. In case of those two countries I would just import the complete South America file http://download.geofabrik.de/south-america.html

P.S. you shouldn't run Nominatim as user `root`

**Comment by maxicabanillas78** [comment]
root@Maxi:~# ./Nominatim/utils/update.php --import-file brazil-latest.osm.pbf
WARNING: resetting cache memory to 502
/usr/bin/osmosis --read-xml '/root/brazil-latest.osm.pbf' --read-empty --derive-change --write-xml-change /root/Nominatim/data/osmosischange.osc
Jan 20, 2015 1:47:38 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Osmosis Version 0.40.1
Jan 20, 2015 1:47:39 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Preparing pipeline.
Jan 20, 2015 1:47:39 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Launching pipeline execution.
Jan 20, 2015 1:47:39 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Pipeline executing, waiting for completion.
Jan 20, 2015 1:47:39 PM org.openstreetmap.osmosis.core.pipeline.common.ActiveTaskManager waitForCompletion
SEVERE: Thread for task 1-read-xml failed
org.openstreetmap.osmosis.core.OsmosisRuntimeException: Unable to parse xml file /root/brazil-latest.osm.pbf.  publicId=(null), systemId=(null), lineNumber=1, columnNumber=1.
        at org.openstreetmap.osmosis.xml.v0_6.XmlReader.run(XmlReader.java:113)
        at java.lang.Thread.run(Thread.java:679)
Caused by: org.xml.sax.SAXParseException: Content is not allowed in prolog.
        at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
        at org.apache.xerces.util.ErrorHandlerWrapper.fatalError(Unknown Source)
        at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
        at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
        at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
        at org.apache.xerces.impl.XMLScanner.reportFatalError(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentScannerImpl$PrologDispatcher.dispatch(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
        at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
        at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at org.apache.xerces.jaxp.SAXParserImpl.parse(Unknown Source)
        at javax.xml.parsers.SAXParser.parse(SAXParser.java:189)
        at org.openstreetmap.osmosis.xml.v0_6.XmlReader.run(XmlReader.java:108)
        ... 1 more
Jan 20, 2015 1:47:39 PM org.openstreetmap.osmosis.core.pipeline.common.ActiveTaskManager waitForCompletion
SEVERE: Thread for task 3-derive-change failed
org.openstreetmap.osmosis.core.OsmosisRuntimeException: An input error has occurred, aborting.
        at org.openstreetmap.osmosis.core.store.DataPostbox.checkForInputErrors(DataPostbox.java:90)
        at org.openstreetmap.osmosis.core.store.DataPostbox.consumeCentralQueue(DataPostbox.java:160)
        at org.openstreetmap.osmosis.core.store.DataPostbox.hasNext(DataPostbox.java:237)
        at org.openstreetmap.osmosis.set.v0_6.ChangeDeriver.run(ChangeDeriver.java:158)
        at java.lang.Thread.run(Thread.java:679)
Jan 20, 2015 1:47:39 PM org.openstreetmap.osmosis.core.Osmosis main
SEVERE: Execution aborted.
org.openstreetmap.osmosis.core.OsmosisRuntimeException: One or more tasks failed.
        at org.openstreetmap.osmosis.core.pipeline.common.Pipeline.waitForCompletion(Pipeline.java:146)
        at org.openstreetmap.osmosis.core.Osmosis.run(Osmosis.java:92)
        at org.openstreetmap.osmosis.core.Osmosis.main(Osmosis.java:37)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:616)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:328)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:408)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:351)
        at org.codehaus.classworlds.Launcher.main(Launcher.java:31)
Error converting osm to osc, osmosis returned: 1

**Comment by mtmail** [comment]
The page lonvia posted also has the answer to that.
You need to execute `--read-pbf` instead of `--read-xml`. You might need to change `update.php` for this to work.

**Comment by LennyLip** [comment]
How to merge two or more pbf files?

**Comment by mtmail** [comment]
May I suggest to ask the question on the https://help.openstreetmap.org/ forum?

This issue is from another user, from January, has a different topic and is closed.

**Comment by LennyLip** [comment]
I know, but google related to this issue for "merge pbf nominatim", may be it will helped sombody.

**Comment by tanzeelrana** [comment]
I accidentally cancelled the process. I do not want to run the whole process again. It took almost a day. Is there a way around it ? I am trying to import Canada's map

**Comment by mtmail** [comment]
@tanzeelrana Please open a new issue. This one got closed 18 months ago. It will help if you can you tell where the processed got stopped (last 10 lines of the output for example).

**Comment by tanzeelrana** [comment]
Thanks for the quick reply, I guess I am luck I did not close the terminal window and still have the last lines where I accidentally cancelled the process. I will open a ticket for this issue. here are a few last lines of the process: 

`  Done 1488793 in 19931 @ 74.697357 per second - Rank 30 ETA (seconds): 37221.117188
  Done 1488803 in 19932 @ 74.694107 per second - Rank 30 ETA (seconds): 37222.601562
  Done 1488811 in 19932 @ 74.694511 per second - Rank 30 ETA (seconds): 37222.292969
  Done 1488823 in 19933 @ 74.691368 per second - Rank 30 ETA (seconds): 37223.699219
  Done 1488836 in 19933 @ 74.692017 per second - Rank 30 ETA (seconds): 37223.203125
  Done 1488848 in 19934 @ 74.688873 per second - Rank 30 ETA (seconds): 37224.609375
  Done 1488860 in 19935 @ 74.685730 per second - Rank 30 ETA (seconds): 37226.011719
  Done 1488899 in 19935 @ 74.687683 per second - Rank 30 ETA (seconds): 37224.519531
  Done 1488944 in 19936 @ 74.686195 per second - Rank 30 ETA (seconds): 37224.656250
  Done 1488999 in 19936 @ 74.688957 per second - Rank 30 ETA (seconds): 37222.542969
  Done 1489057 in 19937 @ 74.688118 per second - Rank 30 ETA (seconds): 37222.187500
  Done 1489109 in 19937 @ 74.690727 per second - Rank 30 ETA (seconds): 37220.187500
  Done 1489151 in 19938 @ 74.689087 per second - Rank 30 ETA (seconds): 37220.445312
  Done 1489196 in 19938 @ 74.691345 per second - Rank 30 ETA (seconds): 37218.714844
  Done 1489248 in 19939 @ 74.690208 per second - Rank 30 ETA (seconds): 37218.585938
  Done 1489304 in 19939 @ 74.693016 per second - Rank 30 ETA (seconds): 37216.437500
  Done 1489362 in 19940 @ 74.692177 per second - Rank 30 ETA (seconds): 37216.078125
  Done 1489379 in 19940 @ 74.693031 per second - Rank 30 ETA (seconds): 37215.425781
  Done 1489409 in 19941 @ 74.690788 per second - Rank 30 ETA (seconds): 37216.140625
  Done 1489457 in 19942 @ 74.689453 per second - Rank 30 ETA (seconds): 37216.164062
  Done 1489502 in 19942 @ 74.691704 per second - Rank 30 ETA (seconds): 37214.441406
  Done 1489534 in 19943 @ 74.689568 per second - Rank 30 ETA (seconds): 37215.078125
  Done 1489585 in 19943 @ 74.692123 per second - Rank 30 ETA (seconds): 37213.121094
  Done 1489639 in 19944 @ 74.691086 per second - Rank 30 ETA (seconds): 37212.914062
  Done 1489684 in 19944 @ 74.693344 per second - Rank 30 ETA (seconds): 37211.187500
  Done 1489740 in 19945 @ 74.692406 per second - Rank 30 ETA (seconds): 37210.906250
^C^C
postgres@tanzeel:~/Documents/osrm/Nominatim/build$ ./utils/setup.php --osm-file ../../data/canada-latest.osm.pbf --all 2>&1 | tee setup.log
tee: setup.log: Permission denied
Create DB
ERROR: database already exists (pgsql://@/nominatim)
database already exists (pgsql://@/nominatim)
postgres@tanzeel:~/Documents/osrm/Nominatim/build$ ./utils/update.php --help
`
The command I was running : 
`./utils/setup.php --osm-file ../../data/map.osm.pbf --all 2>&1 | tee setup.log`

I ran this yesterday. It has been over 24 hours. Is this normal ?
